src/async_impl/client.rs

```diff
@@ -877,6 +887,91 @@ impl ClientBuilder {
             self
         }
     }
+
+    #[cfg(feature = "trust-dns")]
+    /// Sets the name servers for the [trust-dns](trust_dns_resolver) async resolver.
+    ///
+    /// # Example
+    ///
+    /// ``` rust
+    /// use std::net::{IpAddr, Ipv4Addr};
+    /// use reqwest::DnsTransport;
+    ///
+    /// // Use public DNS resolver from Cloudflare:
+    /// let ips = vec![
+    ///     IpAddr::V4(Ipv4Addr::new(1, 1, 1, 1)),
+    ///     IpAddr::V4(Ipv4Addr::new(1, 0, 0, 1)),
+    /// ];
+    /// let client_builder =
+    ///     reqwest::Client::builder().trust_dns_name_servers(&ips, 53, DnsTransport::UdpAndTcp);
+    /// ```
+    ///
+    /// # Optional
+    ///
+    /// This requires the optional `trust-dns` feature to be enabled.
+    pub fn trust_dns_name_servers(
+        mut self,
+        ips: &[IpAddr],
+        port: u16,
+        transport: DnsTransport,
```

Comment:
What makes you opt for passing an enum argument here instead of having separate methods for setting the `name_servers` and setting the transport?

Reply:
As far as I'm concerned, there's no API in `trust-dns-resolver` that allows to change the transport protocol of name servers after initialization.

The methods (from `trust-dns-resolver` crate) that I use to initialize the name servers are the following:

`NameServerConfigGroup::from_ips_clear(ips, port)`

`NameServerConfigGroup::from_ips_tls(ips, port, tls_dns_name)`

`NameServerConfigGroup::from_ips_https(ips, port, tls_dns_name)`

Reply:
Ah yea. So, not saying we have to, but we *could* diverge from the trust-dns API, and allow setting those two steps separately. We don't need to constuct a `NameServerConfigGroup` until `build()`. Does it seem clearer having less arguments per function call, or better to have this extra "security" argument?

Reply:
Indeed, `NameServerConfigGroup` and `ResolverConfig` can be constructed in `ClientBuilder::build` method.
I think that setting ips and transport separately is more consistent with the rest of the `ClientBuilder` methods.

I've updated the PR.

The new API:

``` rust
use std::net::{IpAddr, Ipv4Addr};

let ips = vec![
    IpAddr::V4(Ipv4Addr::new(1, 1, 1, 1)),
    IpAddr::V4(Ipv4Addr::new(1, 0, 0, 1)),
];
let client_builder = reqwest::Client::builder()
    .trust_dns_name_servers(&ips, 853)
    .trust_dns_over_tls("cloudflare-dns.com");
```



Not sure what is the cause of this CI error (https://github.com/seanmonstar/reqwest/pull/900/checks?check_run_id=817836543):

```
C:\Rust\.cargo\bin\cargo.exe build --features blocking,gzip,brotli,json
    Updating crates.io index
error: failed to get `bytes` as a dependency of package `reqwest v0.10.6 (D:\a\reqwest\reqwest)`
##[error]failed to get `bytes` as a dependency of package `reqwest v0.10.6 (D:\a\reqwest\reqwest)`
Caused by:
  failed to load source for dependency `bytes`

Caused by:
  Unable to update registry `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  error inflating zlib stream; class=Zlib (5)
##[error]The process 'C:\Rust\.cargo\bin\cargo.exe' failed with exit code 101
```